addons/event_crm_sale/models/sale_order.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _update_registrations(self, confirm=True, cancel_to_draft=False, registration_data=None):
+        registrations_before = self.env['event.registration'].search([])
+        res = super(SaleOrderLine, self)._update_registrations(confirm, cancel_to_draft, registration_data)
+        registrations_after = self.env['event.registration'].search([])
+        registrations = registrations_after - registrations_before
+        event_crm = self.env['event.crm'].search([])
+        if len(registrations) > 0:
+            if not registrations[0].web_registration:
+                for so_line in self.filtered('event_id'):
+                    for r in event_crm:
+                        r._check_rules(event_id=so_line.event_id, registration_ids=registrations & registrations.search([('event_id','=',so_line.event_id.id)]))
```

Comment:
first build a domain to search only once the registrations based on all so_line.
Then call the _create_leads method for each rules.

Or we could call create_lead, search for all the rules inside the create_leads method, make a dict of lead to create
(adding leads_values for each rules) then call the create only once (need create multi.. if not 'possible' to apply create multi, call create for each lead to create)

Reply:
done